security policy: take 1 (#21119)
vllm-project/vllm: SECURITY.md

# Security Policy

## Reporting security issues:

Please report security issues privately using [the vulnerability submission form](https://github.com/vllm-project/vllm/security/advisories/new).

## Issue triage

Reports will then be triaged by the [vulnerability management team](https://docs.vllm.ai/en/latest/contributing/vulnerability_management.html).

## Threat model

Please see the [Security Guide in the vLLM documentation](https://docs.vllm.ai/en/latest/usage/security.html) for more information on vLLM's security assumptions and recommendations.

Please see [PyTorch's Security Policy](https://github.com/pytorch/pytorch/blob/main/SECURITY.md) for more information and recommendations on how to securely interact with models.

## Issue severity

We will determine the risk of each issue, taking into account our experience dealing with past issues, versions affected, common defaults, and use cases. We use the following severity categories:

> ### CRITICAL Severity:

Reviewer: ```suggestion
### CRITICAL Severity
```

Reviewer: I don't think the colon is necessary since it's already a section heading

Vulnerabilities that allow remote attackers to execute arbitrary code, take full control of the system, or significantly compromise confidentiality, integrity, or availability without any interaction or privileges needed, examples include remote code execution via network, deserialization issues that allow exploit chains. Generally those issues which are rated as CVSS  ≥ 9.0.

> ### HIGH Severity:

Reviewer: ```suggestion
### HIGH Severity
```

Serious security flaws that allow elevated impact—like RCE in specific, limited contexts or significant data loss—but require advanced conditions or some trust, examples include RCE in advanced deployment modes (e.g. multi-node), or high impact issues where some sort of privileged network access is required. These issues typically have CVSS scores between 7.0 and 8.9

> ### MODERATE Severity:

Reviewer: ```suggestion
### MODERATE Severity
```

Vulnerabilities that cause denial of service or partial disruption, but do not allow arbitrary code execution or data breach and have limited impact. These issues have a CVSS rating between 4.0 and 6.9

> ### LOW Severity:

Reviewer: ```suggestion
### LOW Severity
```

Minor issues such as informational disclosures, logging errors, non-exploitable flaws, or weaknesses that require local or high-privilege access and offer negligible impact. Examples include side channel attacks or hash collisions. These issues often have CVSS scores less than 4.0

## Prenotification policy

> For certain security issues of CRITICAL, HIGH, or MODERATE severity level, we may prenotify certain organizations or vendors, which ship vLLM.

Reviewer: ```suggestion
For certain security issues of CRITICAL, HIGH, or MODERATE severity level, we may prenotify certain organizations or vendors that ship vLLM. The purpose of this prenotification is to allow for a coordinated release of fixes for severe issues.
```

> * This prenotification is often in the form of adding their security contacts to the github security advisory, typically few days before release.

Reviewer: ```suggestion
* This prenotification will be in the form of a private email notification. It may also include adding security contacts to the GitHub security advisory, typically a few days before release.
```

I'd like to have the option of not adding people to the advisory, depending on what discussion has taken place there. We can provide the essential information in an email if we don't plan to add them directly. I'm OK with adding them as the default plan, though.

Author: I agree. Also depending on how many people actually want to be added, we may explore several other options. But like you mentioned, lets go with this as the default plan, we can always tweak depending on what kind of issues we need to deal in the future.

* If you wish to be added to the prenotification group, please send an email copying all the members of the [vulnerability management team](https://docs.vllm.ai/en/latest/contributing/vulnerability_management.html). Each vendor contact will be analyzed on a case-by-case basis.

> * We may withdraw notifying certain organizations from future prenotifications if they leak issues before they are public or over time do not add value.

Reviewer: I think we should make it explicit that "leaking" also includes publishing the fix to their customers early. We can specify that a coordinated release date will be specified on the security advisory where coordination is taking place.

> * We may withdraw notifying certain organizations from future prenotifications if they leak issues before they are public or over time do not add value.

Reviewer: ```suggestion
* We may withdraw organizations from receiving future prenotifications if they release fixes or any other information about issues before they are public. Group membership may also change based on policy refinements for who may be included.
```
